docs/dev/building-on-zksync/contracts/contract-development.md

```diff
@@ -0,0 +1,39 @@
+# Smart contract development
+
+zkSync Era allows developers to build projects using the same programming languages and tools used to build on Ethereum.
+We provide our compiler toolchain customized and optimized for development on zkSync. For more information, see
+[the toolchain documentation](../../compiler-toolchain/overview.md).
+
+## EVM compatibility
+
+Almost every smart contract written for EVM will be supported by zkSync Era and will hold all key security invariants so
+that no additional security re-auditing will be required in most cases. A notable exception is the contracts that use
+the following EVM opcodes:
+
+- `SELFDESTRUCT` - It’s considered harmful and was deprecated in [EIP-6049](https://eips.ethereum.org/EIPS/eip-6049).
+- `CALLCODE` - It was deprecated on Ethereum in [EIP-2488](https://eips.ethereum.org/EIPS/eip-2488) in favor of `DELEGATECALL`.
+- `EXTCODECOPY` - We've skipped it for now because zkEVM opcodes are not identical to EVM ones, but it can be implemented if needed.
+- `CODECOPY` - It's replaced with `CALLDATACOPY` in the deploy code.
+- `PC` - Not accessible in Yul and Solidity `>=0.7.0`. Accessible in Solidity `0.6` although it produces a runtime error.
+
+**All these opcodes produce an error on compilation.**
+
+There are several other distinctions; for example, gas metering will be different (as is the case for other L2s as well).
+Some EVM cryptographic precompiles (notably pairings and RSA) won’t be available in the very first release but will be
+implemented soon after the launch, with pairing being a priority to allow both Hyperchains and protocols like
+Aztec/Dark Forest to be deployed without modifications too.
+
+Ethereum cryptographic primitives like `ecrecover`, `keccak256` and `sha256` are supported as precompiles.
+No actions are required from your side as all the calls to the precompiles are done by the compilers under the hood.
+
+### Other considerations
```

Comment:
We should add information about block.timestamp and block.number here

Reply:
It is a different context and important for a reader only coming here and not trying to read more about blocks.

@hedgar2017 I know is beyond the compiler responsibility,. but when it comes to the reader, he doesn't care who is writing it :) 

I think we can simply add a quick info that timestamp and number behaves differently from L1, and for more info a link to the blocks section

Reply:
@githubdoramon it's very hard to document something you don't own :)
I'll add the reference, but even then I'd prefer having a review from Core.

Reply:
Sure.. we can ask them to review (or even myself)

Reply:
Done!
@kailash-manasarovar @StanislavBreadless @vladbochok please check if the reference is sound.